src/v/archival/tests/archival_metadata_stm_gtest.cc

```diff
@@ -200,14 +204,23 @@ TEST_F_CORO(
 
     ASSERT_TRUE_CORO(!res);
 
-    auto [plagued_node, delay_applied] = co_await with_leader(
-      10s, [](raft::raft_node_instance& node) {
-          raft::response_delay fail{
-            .length = 100ms, .on_applied = ss::promise<>{}};
+    ss::shared_promise<> may_resume_append;
+    ss::condition_variable sent_first_append;
 
-          auto delay_applied = fail.on_applied->get_future();
-          node.inject_failure(raft::msg_type::append_entries, std::move(fail));
```

Comment:
nit: I found this declarative syntax slightly easier to use (you don't have create a shared promise and cvar at the call site)